### DISCUSSION TITLE
Iterating the fields of a list view

### DISCUSSION THREAD

**Discussion by web265p3** [pnp/pnpcore#1266] [General] [answered]
I tried already several ways how to do this, but it seems not to be possible to iterate all view fields of a view.

My current approach would be something like this:

 ```
        var newList = _pnPContext.Site.RootWeb.Lists.GetByTitle("Documents");

        await newList.Views.LoadAsync();
        var allItems = newList.Views.AsRequested().FirstOrDefault();
        
        await allItems.LoadAsync(i => i.ViewFields.QueryProperties(v => v.Items));

        foreach (var item in allItems.ViewFields.Items)
        {
            //Do something
        }

```
But QueryProperties does not exist on ViewFields here... is there anything I have overseen to archive my wish?

I am using version 1.10.0 - the last stable build.

**Comment by jansenbe** [comment]
@web265p3 : This works for me:

```csharp
var list = await context.Web.Lists.GetByTitleAsync("Documents", p => p.Title, p => p.ListExperience, p => p.Views.QueryProperties(p => p.ViewFields));
var view = list.Views.AsRequested().FirstOrDefault();

foreach (var field in view.ViewFields.Items)
{

}
```

**Comment by web265p3** [comment]
Thanks,
I am using the last stable build - which version are you using? 
If I use your code, I get a null ref error: 

![image](https://github.com/pnp/pnpcore/assets/20642406/0bc6566a-e0b0-42e0-937d-0c58919509be)

**Reply by jansenbe** [reply, marked as answer]
@web265p3 : you need to load the views when you call `GetByTitle`

**Reply by web265p3** [reply]
Thanks, this works.
I am just wondering why the line "await view.LoadAsync(v => v.ViewFields);" before my call did not also work...

**Reply by darioniz** [reply]
Can you post a full working sample? I am also having a problem loading ViewFields.Items.
